src/Model/GServer.php

```diff
@@ -1356,14 +1358,15 @@ private static function fetchWeeklyUsage(string $url, array $serverdata): array
 	private static function detectFromContacts(string $url, array $serverdata): array
 	{
 		$gserver = DBA::selectFirst('gserver', ['id'], ['nurl' => Strings::normaliseLink($url)]);
-		if (empty($gserver)) {
-			return $serverdata;	
+
+		if (!DBA::isResult($gserver)) {
+			return $serverdata;
 		}
 
 		$contact = Contact::selectFirst(['id'], ['uid' => 0, 'failed' => false, 'gsid' => $gserver['id']]);
 
 		// Via probing we can be sure that the server is responding
-		if (!empty($contact['id']) && Contact::updateFromProbe($contact['id'])) {
+		if (DBA::isResult($contact) && ($contact['id'] > 0) && Contact::updateFromProbe($contact['id'])) {
```

Comment:
`empty(...)` does the whole job.

Reply:
It didn't seem to catch it. See https://github.com/friendica/friendica/issues/11630#issuecomment-1164883584= for a backtrace where `null` was handled to `Contact::updateFromProbe()`.

Reply:
Possibly php performs both checks in that `if`. So splitting it into two checks would be a better solution. `empty(...)` is really reliable. We use it all the time. I also want to replace all the `DBA::isResult(...)` calls in the future with `empty(...)` checks since that function doesn't serve any additional purpose.

Reply:
Folks, @copiis is running an old version of `GServer.php` from April 2nd where there's no `empty()` check yet, see line number in the original report and compare with this version: https://github.com/friendica/friendica/blob/e299fc67c8809ecf1018e6680a3a5284bcf7d8d9/src/Model/GServer.php#L1353

It was fixed in https://github.com/friendica/friendica/blob/62367923e069da9fe6e0d8bde741bdb3360610cd/src/Model/GServer.php#L1357 and the line number changes.

Reverting this change is correct.
```suggestion
		if (!empty($contact['id']) && Contact::updateFromProbe($contact['id'])) {
```